### PR TITLE
feat(core): cors config

### PR DESCRIPTION
**Pull request by pflynn-virtru**
Introduces configuration via `opentdf.yaml` for CORS

```yaml
  cors:
    enabled: false
    # '*' to allow any origin or a specific domain like 'https://yourdomain.com'
    allowedorigins: '*'
    # List of methods. Examples: 'GET,POST,PUT'
    allowedmethods:
      - GET
      - POST
      - PATCH
      - PUT
      - DELETE
      - OPTIONS
    # List of headers that are allowed in a request
    allowedheaders:
      - ACCEPT
      - Authorization
      - Content-Type
      - X-CSRF-Token
    # List of response headers that browsers are allowed to access
    exposedheaders:
      - Link
    # Sets whether credentials are included in the CORS request
    allowcredentials: true
    # Sets the maximum age (in seconds) of a specific CORS preflight request
    maxage: 3600
```

resolves https://github.com/opentdf/platform/issues/305

### CODE REVIEW

File: opentdf-with-hsm.yaml [hunk 1]
```diff
@@ -71,7 +71,31 @@ server:
       #
       #  [matchers]
       #  m = g(r.sub, p.sub) && globOrRegexMatch(r.res, p.res) && globOrRegexMatch(r.act, p.act) && globOrRegexMatch(r.obj, p.obj)
-
+  cors:
+    enabled: false
+    # '*' to allow any origin or a specific domain like 'https://yourdomain.com'
+    allowedorigins: '*'
```
Comment (dmihalcik) [inline opentdf-with-hsm.yaml:77]: Is there a reason these aren't lower camelCased?

File: opentdf-dev.yaml [hunk 2]
```diff
@@ -66,7 +67,31 @@ server:
       #
       #  [matchers]
       #  m = g(r.sub, p.sub) && globOrRegexMatch(r.res, p.res) && globOrRegexMatch(r.act, p.act) && globOrRegexMatch(r.obj, p.obj)
-
+  cors:
+    enabled: false
+    # '*' to allow any origin or a specific domain like 'https://yourdomain.com'
+    allowedorigins: '*'
```
Comment (dmihalcik-virtru) [inline opentdf-dev.yaml:73]: Shouldn't these be lower camelCase?
```suggestion
    allowedOrigins: '*'
```

**Review by patmantru — APPROVED** [review]

File: opentdf-dev.yaml (hunk 2 above)
Comment (pflynn-virtru) [reply opentdf-dev.yaml:73]: Viper does not allow camelcase https://github.com/opentdf/platform/issues/745

File: opentdf-example.yaml [hunk 3]
```diff
@@ -66,7 +66,31 @@ server:
       #
       #  [matchers]
       #  m = g(r.sub, p.sub) && globOrRegexMatch(r.res, p.res) && globOrRegexMatch(r.act, p.act) && globOrRegexMatch(r.obj, p.obj)
-
+  cors:
+    enabled: false
+    # '*' to allow any origin or a specific domain like 'https://yourdomain.com'
+    allowedorigins: '*'
```
Comment (jrschumacher) [inline opentdf-example.yaml:72]: @pflynn-virtru having a wildcard in the example is a footgun IMO. I would advise using `localhost:*` if we wanted any kind of default value that would be convenient for developers.

**Pull request merged**